Zend/zend_execute.c

```diff
@@ -2369,6 +2369,7 @@ static zend_never_inline zval* ZEND_FASTCALL zend_find_array_dim_slow(HashTable
 		hval = 1;
 		goto num_idx;
 	} else if (Z_TYPE_P(offset) == IS_RESOURCE) {
+		zend_error(E_WARNING, "Resource ID#%d used as offset, casting to integer (%d)", Z_RES_HANDLE_P(offset), Z_RES_HANDLE_P(offset));
```

Comment:
was this an omission here?

Reply:
This is part of isset() code, so it's not entirely unusual for some warnings to be suppressed. However, I think this particular one should indeed be thrown, because it may be generated even if isset() returns true (while the ones we may suppress only happen if it returns false).

This should probably use `zend_use_resource_as_offset()` though.

This also needs test coverage (best add to some existing test that checks isset with different types), and possibly also a corresponding adjustment in JIT code.